### ISSUE TITLE
pipe-connector is returning "Invalid Argument" 17692.1004

### ISSUE THREAD

**Issue by AngellusMortis** [rupor-github/ssh-agent-wsl#2]
I am on the latest 1803 Slow Ring build, 17692.1004. 

Whenever I try to start `ssh-agent-wsl`, I get the error `pipe-connector.exe: Invalid argument`

![image](https://user-images.githubusercontent.com/490848/42326305-c6b88df2-8036-11e8-97b2-38815ead8f01.png)

**Comment by rupor-github** [comment]
As specified in documentation:

Download the archive from the releases page and unpack it in a convenient location on the Windows part of your drive. Because WSL can only execute Win32 binaries from drvfs locations, ssh-agent-wsl will not work if unpacked inside the WSL filesystem (onto an lxfs mount). 

You placed pipe-connector.exe on lxfs mount, this will always result in:

ssh-agent-wsl: win32 helper died immediately
error fetching identities: communication with agent failed

Please, move it somewhere to /mnt/{c|e|...) - it will work. I have it like this:

```
l /usr/local/bin
total 64M
drwxr-xr-x   1 root root  512 Jun 21 20:07 .
drwxr-xr-x   1 root root  512 Jun  6 07:56 ..
-rwxr-xr-x 123 root root  14M Jun 21 20:07 git
-rwxr-xr-x   2 root root 159K Jun 21 20:07 git-cvsserver
-rwxr-xr-x 123 root root  14M Jun 21 20:07 git-receive-pack
-rwxr-xr-x   2 root root 7.7M Jun 21 20:07 git-shell
-rwxr-xr-x 123 root root  14M Jun 21 20:07 git-upload-archive
-rwxr-xr-x 123 root root  14M Jun 21 20:07 git-upload-pack
lrwxrwxrwx   1 root root   42 May  5 20:11 pipe-connector.exe -> /mnt/c/Users/rupor/.wsl/pipe-connector.exe
-rwxr-xr-x   1 root root 1.3K Apr 27 16:59 ranger
-rwxr-xr-x   1 root root  18K Apr 27 16:59 rifle
-rwxr-xr-x   1 root root  27K May  5 20:10 ssh-agent-wsl
-rwxr-xr-x   1 root root 646K Jun  9 09:28 tmux
lrwxrwxrwx   1 root root   37 May  9 13:12 win32yank -> /mnt/c/Users/rupor/.wsl/win32yank.exe
-rwxr-xr-x   2 root root 818K May 25 13:53 zsh
-rwxr-xr-x   2 root root 818K May 25 13:53 zsh-5.5.1
```

**Comment by AngellusMortis** [comment]
You  can see above where my `pipe-connector.exe` is. It is **not** on a lxfs mount. It is in `C:\Windows\System32` (`/mnt/c/Windows/System32`)

EDIT: Same thing happens even if I reinstall `ssh-agent-wsl` and `pipe-connector.exe` to a new location, like `/mnt/b/bin` or if I put `ssh-agent-wsl` direct on a lxfs mount and synlink to `pipe-connector.exe` like you do.

![image](https://user-images.githubusercontent.com/490848/42331679-09a653a8-8044-11e8-8012-55e7f142c829.png)

**Comment by rupor-github** [comment]
My bad, that however works on my system (1803 - 17134.137):

```
$ eval $(ssh-agent-wsl -H /mnt/c/Windows/System32/pipe-connector.exe -r)
ssh-agent-wsl pid 270

# rupor at Bionic in ~ [11:02:51]
$ ssh-add -l
<<<<properly prints what it has to>>>
```
you could pass debug flag to ssh-agent-wsl: `ssh-agent-wsl -d` and use something like dbgview.exe from Sysinternals to see pipe-connector debug trace, not sure how useful it may be.  

Does it work when pipe-connector.exe placed outside of System32 directory? Looks like Microsoft is playing with security settings in the latest builds. I do not (and not planning to) install insiders builds - no time for that...

**Comment by AngellusMortis** [comment]
>  (1803 - 17134.137):

I know, it literally worked yesterday. I just got 17692.1004 today.

> Does it work when pipe-connector.exe placed outside of System32 directory?

No, in my example above, I tried that. I thought that might have been an issue. 

Here are the [release notes for 17692](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17692). I am not 100% sure how `ssh-agent-wsl` works, but if you look at 17704, this issue might be resolved:

> Non-blocking Unix sockets failing indefinitely with EAGAIN [GH 3191]

17704 is only Fast Ring right now though so I do not have it. 17692 came out to Slow Ring 3 days ago.

**Comment by rupor-github** [comment]
In the current ssh implementation Microsoft is not using UNIX sockets - named pipe is being used, which makes sense from security standpoint. So unless they switched Windows open ssh to proper sockets - this probably has nothing to do with issue at hand. I looked at https://github.com/PowerShell/openssh-portable and I do not see any relevant changes (may be they did not commit them yet or I am missing something). If however they moved - you do not need ssh-agent-wsl at all, standard agent should be able to properly connect to Windows ssh-agent service...

**Comment by tuhlmann** [comment]
weasel-pageant has the same issue on the latest insider builds. It works fine on my 1803 build and failed when I installed build 17711 (I can't comment on insider builds before that as I switched from stable directly to this one).
I thought I try ssh-agent-wsl instead which also fails. Since it's forked from the weasel code base chances are it's the same issue: https://github.com/vuori/weasel-pageant/issues/5
